src/App.vue

```diff
@@ -1,8 +1,132 @@
 <script lang="ts" setup>
+const toggleDrawer = () => {
+  const drawer = document.querySelector('.nav-links')
+  drawer?.classList.toggle('active')
+}
+
+const reloadPage = () => {
+  location.reload()
+}
+
+const scrollToElement = (id: string) => {
+  const drawer = document.querySelector('.nav-links')
+  drawer?.classList.toggle('active')
+
+  document.getElementById(id)?.scrollIntoView()
+}
 </script>
 
 <template>
+  <header h-12 md:h-20 lg:h-28 flex flex-row justify-between items-center gap-2 lg:gap-4 mx-0 class="header">
+    <button mx-4 md:mx-8 class="hamburger" :onclick="toggleDrawer">
+      <div class="bar" />
+    </button>
+
+    <div flex flex-row items-center gap-2>
+      <img w-8 md:w-12 lg:w-16 src="./assets/logo.png" alt="Logo">
+      <h1 text-white text-xl md:text-3xl font-bold>
+        超級智慧潛水
+      </h1>
+    </div>
+
+    <div w-16 />
+  </header>
+
+  <nav class="nav-links">
+    <img w-20 md:w-28 lg:w-40 mx-auto my-4 md:my-8 src="./assets/login.png" alt="Login">
+    <ul text-base md:text-xl font-600 md:font-900 class="nav-list">
+      <li :onclick="reloadPage">
+        <img w-8 md:w-10 src="./assets/home.png" alt="Home">
+        <div>首頁</div>
+      </li>
+      <li :onclick="toggleDrawer">
+        <img w-8 md:w-10 src="./assets/dark.png" alt="Dark Mode">
+        <div>深色主題</div>
+      </li>
+      <li :onclick="() => scrollToElement('uv-chart')">
+        <img w-8 md:w-10 src="./assets/chart.png" alt="UV Chart">
+        <div>紫外線指數</div>
+      </li>
+      <li :onclick="() => scrollToElement('temperature-chart')">
+        <img w-8 md:w-10 src="./assets/chart.png" alt="Temperature Chart">
+        <div>氣溫與水溫</div>
+      </li>
+      <li :onclick="() => scrollToElement('ocean-depth-chart')">
+        <img w-8 md:w-10 src="./assets/chart.png" alt="Ocean Depth Chart">
+        <div>水下深度</div>
+      </li>
+      <li :onclick="() => scrollToElement('wind-chart')">
+        <img w-8 md:w-10 src="./assets/chart.png" alt="Wind Chart">
+        <div>風向與風速</div>
+      </li>
+    </ul>
+  </nav>
+
   <main font-sans h-full p="x-4 y-4" text="center gray-700">
     <router-view />
   </main>
 </template>
+
+<style scoped lang="scss">
```

Comment:
Suggest removing css (are they `scss` ?) here and move them into their own component file.
App.vue is always recommended to remain cleaned.

Reply:
Addressed at ee59e7a.